.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,100 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+name: Documentation
+
+on: 
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
+  links-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Set Up Env
+        run: make env
+      
+      - name: Install pandoc
+        run: |
+          sudo apt-get update
+          sudo apt-get install pandoc
+
+      - name: Build docs
+        run: make docs
+
+      # NOTE: 
+      #   I was not able to make "lycheeverse/lychee-action@v1.2.0" work, 
+      #   My main problem was that the glob patterns (e.g. "./docs/build/html/**/*.html") did not work.
+      #
+      #   In order to use the glob pattern properly we need to surround it with double quotes (") but 
+      #   when we do this, some part of the logic that handles the 'with.args' key, additionally wraps 
+      #   it into ordinary quotes ('), as a result lychee binary is executed as - lychee '"glob/**/*.pattern"'
+      #
+      #   I decided to overcome this with help of a custom script (take a look at the job step below)
+      #
+      #   Also, please take a look at this two related issues in the lychee-action repo
+      #   - https://github.com/lycheeverse/lychee-action/issues/68
+      #   - https://github.com/lycheeverse/lychee-action/issues/67
+      #
+      # - name: Link Checker
+      #   uses: lycheeverse/lychee-action@v1.2.0
+      #   with:
+      #     args: --exclude-file docs/.lycheeignore --verbose --exclude-loopback --exclude-mail --no-progress ./docs/build/html/**/*.html
+      #     fail: true
+      #   env:
+      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+      - name: Link Checker
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        run: |
+          set -uxo pipefail
+          
+          curl \
+            -L --output lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz \
+            https://github.com/lycheeverse/lychee/releases/download/v0.8.2/lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz
+          
+          tar -xvzf lychee-v0.8.2-x86_64-unknown-linux-gnu.tar.gz
+
+          echo "$GITHUB_TOKEN"
```

Comment:
Please remove this

Reply:
OMG, sorry, I have been testing it locally and forgot to remove this